src/metrics.rs

```diff
@@ -2607,50 +2602,61 @@ impl GooseAttack {
             // Prepare report summary variables.
             let users = self.metrics.users.to_string();
 
-            let starting = self.metrics.starting.unwrap();
-            let started = if self.metrics.started.is_some() {
-                self.metrics.started.unwrap()
-            } else {
-                self.metrics.stopping.unwrap()
-            };
-            let (starting_seconds, starting_minutes, starting_hours) =
-                self.metrics.get_seconds_minutes_hours(&starting, &started);
-            let stopping = self.metrics.stopping.unwrap();
-            let (running_seconds, running_minutes, running_hours) =
-                self.metrics.get_seconds_minutes_hours(&started, &stopping);
-            let stopped = self.metrics.stopped.unwrap();
-            let (stopping_seconds, stopping_minutes, stopping_hours) =
-                self.metrics.get_seconds_minutes_hours(&stopping, &stopped);
-
-            let mut report_range = format!(
-                "<p>Starting: <span>{} - {} (Duration: {:02}:{:02}:{:02})</span></p>",
-                starting.format("%Y-%m-%d %H:%M:%S"),
-                started.format("%Y-%m-%d %H:%M:%S"),
-                starting_hours,
-                starting_minutes,
-                starting_seconds,
-            );
-
-            if self.metrics.started.is_some() {
-                report_range.push_str(&format!(
-                    "<p>Running: <span>{} - {} (Duration: {:02}:{:02}:{:02})</span></p>",
-                    started.format("%Y-%m-%d %H:%M:%S"),
-                    stopping.format("%Y-%m-%d %H:%M:%S"),
-                    running_hours,
-                    running_minutes,
-                    running_seconds,
-                ));
+            let mut steps_overview = String::new();
+            for step in self.metrics.history.windows(2) {
```

Comment:
This loop is basically a copy-paste of the same thing in console printout part. Maybe we should move it to a helper and try to share it?